databricks/koalas/frame.py

```diff
@@ -3494,6 +3494,116 @@ def append(self, other: 'DataFrame', ignore_index: bool = False,
+        errors : {'raise', 'ignore'}, default 'ignore'
+            If 'raise', will raise a ValueError if the DataFrame and `right` both contain non-NA
+            data in the same place.
+
+        Returns
+        -------
+        None : method directly changes calling object
+
+        Raises
+        ------
+        ValueError
+            * When `errors='raise'` and there's overlapping non-NA data.
+            * When `errors` is not either 'ignore' or 'raise'
+
+        See Also
+        --------
+        DataFrame.merge : For column(s)-on-columns(s) operations.
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'A': [1, 2, 3], 'B': [400, 500, 600]}, columns=['A', 'B'])
+        >>> new_df = ks.DataFrame({'B': [4, 5, 6], 'C': [7, 8, 9]}, columns=['B', 'C'])
+        >>> df.update(new_df)
+        >>> df
+           A  B
+        0  1  4
+        1  2  5
+        2  3  6
+
+        The DataFrame's length does not increase as a result of the update,
+        only values at matching index/column labels are updated.
+
+        >>> df = ks.DataFrame({'A': ['a', 'b', 'c'], 'B': ['x', 'y', 'z']}, columns=['A', 'B'])
+        >>> new_df = ks.DataFrame({'B': ['d', 'e', 'f', 'g', 'h', 'i']}, columns=['B'])
+        >>> df.update(new_df)
+        >>> df
+           A  B
+        0  a  d
+        1  b  e
+        2  c  f
+
+        For Series, it's name attribute must be set.
+
+        >>> df = ks.DataFrame({'A': ['a', 'b', 'c'], 'B': ['x', 'y', 'z']}, columns=['A', 'B'])
+        >>> new_column = ks.Series(['d', 'e'], name='B', index=[0, 2])
+        >>> df.update(new_column)
+        >>> df
+           A  B
+        0  a  d
+        1  b  y
+        2  c  e
+
+        If `other` contains None the corresponding values are not updated in the original dataframe.
+
+        >>> df = ks.DataFrame({'A': [1, 2, 3], 'B': [400, 500, 600]}, columns=['A', 'B'])
+        >>> new_df = ks.DataFrame({'B': [4, None, 6]}, columns=['B'])
+        >>> df.update(new_df)
+        >>> df
+           A      B
+        0  1    4.0
+        1  2  500.0
+        2  3    6.0
+        """
+        if errors not in ['ignore', 'raise']:
+            raise ValueError("The parameter errors must be either 'ignore' or 'raise'")
+
+        if join != 'left':
+            raise NotImplementedError("Only left join is supported")
+
+        if isinstance(other, ks.Series):
+            other = DataFrame(other)
+
+        update_columns = list(set(self._internal.data_columns)
+                              .intersection(set(other._internal.data_columns)))
+        update_sdf = self.join(other[update_columns], rsuffix='_new')._sdf
+
+        for column_name in update_columns:
+            old_col = update_sdf[column_name]
+            new_col = update_sdf[column_name + '_new']
+            if errors == 'raise':
+                null_sdf = update_sdf.filter(old_col.isNull() & new_col.isNull())
```

Comment:
@yuwillrun, seems https://github.com/databricks/koalas/pull/481/files#r296667059 comment was missed.

`update_sdf.filter(old_col.isNull() & new_col.isNull())` always return a dataframe. So, it will always raise an exception.

If it's tricky to do, we can remove `errors` for now and do it later in another PR.

Reply:
Sorry, i will check it now.

Reply:
Fixed already.